luaui/Widgets/gui_factionpicker.lua

```diff
@@ -16,14 +16,40 @@ local factions = {}
 if UnitDefNames.dummycom then
 	factions[#factions+1] = { startUnit = UnitDefNames.dummycom.id, faction = 'random' }
 end
-if UnitDefNames.corcom then
-	factions[#factions+1] = { startUnit = UnitDefNames.corcom.id, faction = 'cor' }
+
+local allyTeamBitmask = 0
+do
+	local factionlimiter = tonumber(Spring.GetModOptions().factionlimiter)
+	if factionlimiter and factionlimiter > 0 then
+		local allyTeamID = Spring.GetMyAllyTeamID()
+		factionlimiter = math.floor(factionlimiter/2^(allyTeamID*3))
+		allyTeamBitmask = math.bit_and(factionlimiter, 7)
+	end
+end
+if allyTeamBitmask == 0 then
+	allyTeamBitmask = 7
 end
-if UnitDefNames.armcom then
-	factions[#factions+1] = { startUnit = UnitDefNames.armcom.id, faction = 'arm' }
+
+local function addOptions(bitmask)
```

Comment:
Personally I'd put the list in team rules params rather than require widgets to recalculate it via bitmasks on their own